app/controllers/application_controller.rb

```diff
@@ -1,2 +1,17 @@
 class ApplicationController < ActionController::Base
+  # protect_from_forgery prepend: true
+
+  before_action :authenticate_user!
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :email, :password, :password_confirmation) }
+    devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:name, :email, :password, :current_password) }
+  end
+
+  def after_sign_up_path_for(_resource)
+    root_path
+  end
```

Comment:
### OPTIONAL
- [ ] When ı try to sign in with the wrong user information ı am not get any message. It will be good if you add some flash messages for that.